vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SnapshotManager.java

```diff
@@ -0,0 +1,223 @@
+import edu.asu.diging.vspace.core.file.IStorageEngine;
+import edu.asu.diging.vspace.core.model.ISnapshotTask;
+import edu.asu.diging.vspace.core.model.impl.ExhibitionSnapshot;
+import edu.asu.diging.vspace.core.model.impl.SequenceHistory;
+import edu.asu.diging.vspace.core.model.impl.SnapshotTask;
+import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.model.impl.SpaceStatus;
+import edu.asu.diging.vspace.core.services.IRenderingManager;
+import edu.asu.diging.vspace.core.services.ISnapshotManager;
+
+@Service
+@PropertySource("classpath:app.properties")
+public class SnapshotManager  implements  ISnapshotManager {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Value("${page_size}")
+    private int pageSize;
+    
+    @Value("${resources_folder_path}")
+    private String resourcesPath;
+
+    @Autowired
+    @Qualifier("storageEngineDownloads")
+    private IStorageEngine storageEngineDownloads;
+
+    @Autowired
+    private IRenderingManager renderingManager;
+
+    private SequenceHistory sequenceHistory;
+
+    @Autowired
+    private ExhibitionSnapshotRepository exhibitionSnapshotRepository;
+
+    @Autowired
+    private SnapshotTaskRepository snapshotTaskRepository;
+    
+    @Autowired
+    private SpaceRepository spaceRepository;
+    
+    private final String RESOURCES_FOLDER_NAME = "resources";
+    private final String ZIP_FILE_EXTENSION = ".zip";
+
+    /**
+     * Triggers the creation of an exhibition snapshot.
+     *
+     * This method initializes and saves an {@link ExhibitionSnapshot} and a {@link SnapshotTask},
+     *
+     * @return the created {@link ExhibitionSnapshot}
+     * @throws IOException                          if an I/O error occurs during the snapshot creation process
+     * @throws InterruptedException                 if the snapshot creation process is interrupted
+     * @throws SnapshotCouldNotBeCreatedException   if the snapshot could not be created due to any other errors
+     */
+    @Override
+    @Transactional
+    public ExhibitionSnapshot triggerExhibitionSnapshotCreation() throws IOException, InterruptedException, SnapshotCouldNotBeCreatedException {
+        String exhibitionFolderName = getExhibitionFolderName();
+        ExhibitionSnapshot exhibitionSnapshot = new ExhibitionSnapshot();
+        createSnapshotFolder(exhibitionSnapshot, exhibitionFolderName);       
+        SnapshotTask snapshotTask =  createSnapshotTask(exhibitionSnapshot);
+        exhibitionSnapshot.setSnapshotTask(snapshotTask); 
+        exhibitionSnapshot = exhibitionSnapshotRepository.save(exhibitionSnapshot);
+
+        try {
+            snapshotTask = createSnapshot(resourcesPath, exhibitionFolderName, sequenceHistory, exhibitionSnapshot);
+            storageEngineDownloads.generateZip(exhibitionFolderName);           
+            snapshotTask.setTaskComplete(true);
```

Comment:
wait, I'm confused. `createSnapshot` spins up a new thread, right? so you can't generate a zip file right after you call it, can you? and it also wont' be complete right after, doesn't it? Or am I missing something?

Reply:
wait, actually, if this method calls `createSnapshot`, it won't spin up a new thread. It should though, since it could take quite a while.